### PR TITLE
feat(space-nuxt-base): add webhook update logic

### PR DESCRIPTION
**Pull request by demetriusfeijoo**
## What?

- adds webhook update logic
- does some refactoring in the whole webhook code

## Why?

JIRA: EXT-3181

I think this may help to review it more easily:
Screenshot 2024-05-23 at 11 29 31

### CODE REVIEW

File: space-plugins/nuxt-base/server/utils/webhook.ts [hunk 1]
```diff
@@ -1,20 +1,76 @@
...
 };
 
-export const createStoryblokWebhook: CreateStoryblokWebhook = async (
+export const createStoryblokWebhook: CreateStoryblokWebhook = async (params) =>
+	saveStoryblokWebhook('create', params);
+
+export const updateStoryblokWebhook: UpdateStoryblokWebhook = async (params) =>
+	saveStoryblokWebhook('update', params);
+
+export const deleteStoryblokWebhookById: DeleteStoryblokWebhook = async (
```
Comment (demetriusfeijoo) [inline space-plugins/nuxt-base/server/utils/webhook.ts:22]: It was just moved, nothing changed.

File: space-plugins/nuxt-base/server/utils/webhook.ts [hunk 2]
```diff
@@ -1,20 +1,76 @@
...
+	} else if (body.user_id) {
+		return 'user';
+	} else if (body.asset_id) {
+		return 'asset';
+	} else if (body.story_id) {
+		return 'story';
+	}
+};
+
+const saveStoryblokWebhook: SaveStoryblokWebhook = async (
```
Comment (BibiSebi) [inline space-plugins/nuxt-base/server/utils/webhook.ts:69]: nit: could we rename this to `upsertStoryblokWebhook`, it might be more fitting IMO.

Comment (demetriusfeijoo) [reply space-plugins/nuxt-base/server/utils/webhook.ts:69]: Thanks for the suggestion, Bibi.
I was trying to think of a better name but with no luck. 🤣

Comment (demetriusfeijoo) [reply space-plugins/nuxt-base/server/utils/webhook.ts]: Done Bibi https://github.com/storyblok/space-tool-plugins/pull/56/commits/93f0fb3d5434570f6db1e06b0ad110c43ffdcd60
🙌

**Review by eunjae-lee — APPROVED** [review]
awesome!

**Review by Dawntraoz — APPROVED** [review]
LGTM! 🚀

**Pull request merged**